File: Practice/src/exercise/CollatzConjecture.py
def solution(num, count=0):
    if num == 1:
        return count
    elif count > 500:
        return -1
    else:
        return solution(num * 3 + 1 if num % 2 else num / 2, count + 1)
    # for i in range(500):
    #     if num == 1:
    #         return i
    #     num = num * 3 + 1 if num % 2 else num / 2
    # return -1

File: Practice/src/exercise/CollatzConjecture.js
function solution(num) {
    for (let i = 0; ; i++) {
        if (i > 500) return -1;
        else if (num == 1) return i;
        num = num % 2 == 0 ? num / 2 : num * 3 + 1;
    }
}
